internal/testing/s3_test.go

```diff
@@ -130,6 +134,51 @@ func TestS3(t *testing.T) {
 	}
 }
 
+func TestS3Authentication(t *testing.T) {
```

Comment:
can we extend the test with an authenticated client to make the test more complete?